plugins/extraction/extract_env.ml

```diff
@@ -135,19 +135,22 @@ let check_arity env cb =
   let t = cb.const_type in
   if Reduction.is_arity env t then raise Impossible
 
+let get_body lbody =
+  EConstr.of_constr (Mod_subst.force_constr lbody)
+
 let check_fix env cb i =
   match cb.const_body with
     | Def lbody ->
-	(match Constr.kind (Mod_subst.force_constr lbody) with
-	  | Fix ((_,j),recd) when Int.equal i j -> check_arity env cb; (true,recd)
+        (match EConstr.kind Evd.empty (get_body lbody) with
+          | Fix ((_,j),recd) when Int.equal i j -> check_arity env cb; (true,recd)
 	  | CoFix (j,recd) when Int.equal i j -> check_arity env cb; (false,recd)
 	  | _ -> raise Impossible)
     | Undef _ | OpaqueDef _ -> raise Impossible
 
 let prec_declaration_equal (na1, ca1, ta1) (na2, ca2, ta2) =
   Array.equal Name.equal na1 na2 &&
-  Array.equal Constr.equal ca1 ca2 &&
-  Array.equal Constr.equal ta1 ta2
+  Array.equal (EConstr.eq_constr Evd.empty) ca1 ca2 &&
```

Comment:
`Evd.empty` should not even exist, please use `Evd.from_env env` where env is the global one if you don't want to thread it around.  The point is that `Evd` today also contains universes as in *universe name* and *universe constraints*. I think you probably don't care about the latter, but some code may `assert false` if it faces a universe name not part of the evar map.

Why `Evd.empty` has not *big fat warning* attached to is a mystery to me CC @ppedrot

Reply:
> Why Evd.empty has not big fat warning attached to is a mystery to me CC @ppedrot

Yes! Can we put this in 8.8? In fact as it is used in so many parts of the codebase we tend to think it is OK.